src/machine/usb/adc/midi/midi.go

```diff
@@ -79,12 +96,16 @@ func (m *midi) sendUSBPacket(b []byte) {
 }
 
 // from BulkIn
-func (m *midi) Handler() {
+func (m *midi) TxHandler() {
 	m.waitTxc = false
 	if b, ok := m.buf.Get(); ok {
 		m.waitTxc = true
 		m.sendUSBPacket(b)
 	}
+
+	if m.txHandler != nil {
+		m.txHandler()
+	}
```

Comment:
Oops, I made a mistake.

`func (m *midi) TxHandler()` is called when a previous USB transmission has completed.
Line 103 is the next transmission.

Therefore, m.txHandler() should be called immediately after entering this function.

Reply:
Calling immediately after entering a function can add a little more data for the next transmission.
If the call is made at the end of a function, USB communication may improve performance slightly, but only by a small percentage.

So it is probably better to call from the beginning of the function this time.

Reply:
OK change made :smiley_cat: 

Thanks for helping look this over @sago35 it is appreciated!